### PR TITLE
Armor.json sorting pt. 6

### PR DESCRIPTION
**Pull request by RarkGrames**
#### Summary
```SUMMARY: Infrastructure "Sorted items out of armor.json into separate files"```

#### Purpose of change
Sorting out armor.json for easier handling.
Continuation of #37441.

#### Describe the solution
Sorting applicable items to tool_armor.json and undergarment.json. Moving the things I missed earlier.

#### Testing
Game loads properly, line counts check out.

### CODE REVIEW

File: data/json/items/armor/undergarment.json [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+    "covers": [ "TORSO" ],
+    "coverage": 75,
+    "encumbrance": 15,
+    "warmth": 25,
+    "material_thickness": 3,
+    "environmental_protection": 1,
+    "flags": [ "VARSIZE", "FANCY", "SKINTIGHT" ]
+  },
+  {
+    "id": "hot_pants",
```
Comment (I-am-Erk) [inline data/json/items/armor/undergarment.json]: Hot pants aren't undergarments

File: data/json/items/armor/undergarment.json [hunk 2]
```diff
@@ -0,0 +1,743 @@
...
+    "symbol": "[",
+    "looks_like": "boxer_briefs",
+    "color": "light_red",
+    "covers": [ "LEGS" ],
+    "coverage": 15,
+    "material_thickness": 1,
+    "flags": [ "VARSIZE", "WATER_FRIENDLY", "SKINTIGHT" ]
+  },
+  {
+    "id": "hot_pants_fur",
```
Comment (I-am-Erk) [inline data/json/items/armor/undergarment.json]: see above

File: data/json/items/armor/undergarment.json [hunk 3]
```diff
@@ -0,0 +1,743 @@
...
+    "looks_like": "hot_pants_leather",
+    "color": "brown",
+    "covers": [ "LEGS" ],
+    "coverage": 15,
+    "warmth": 10,
+    "material_thickness": 1,
+    "flags": [ "VARSIZE", "SKINTIGHT" ]
+  },
+  {
+    "id": "hot_pants_leather",
```
Comment (I-am-Erk) [inline data/json/items/armor/undergarment.json]: see above

File: data/json/items/armor/undergarment.json [hunk 4]
```diff
@@ -0,0 +1,743 @@
...
+    "color": "light_gray",
+    "covers": [ "LEGS" ],
+    "coverage": 60,
+    "encumbrance": 7,
+    "warmth": 20,
+    "material_thickness": 2,
+    "flags": [ "VARSIZE", "SKINTIGHT", "OVERSIZE" ]
+  },
+  {
+    "id": "leggings",
```
Comment (I-am-Erk) [inline data/json/items/armor/undergarment.json]: Leggings probably don't belong in undergarments either, although personally I feel fashion could learn a thing or two from this.

File: data/json/items/armor/undergarment.json [hunk 5]
```diff
@@ -0,0 +1,743 @@
...
+    "looks_like": "leg_warmers",
+    "color": "dark_gray",
+    "covers": [ "LEGS" ],
+    "coverage": 100,
+    "warmth": 20,
+    "material_thickness": 1,
+    "flags": [ "VARSIZE", "SKINTIGHT", "WATER_FRIENDLY" ]
+  },
+  {
+    "id": "loincloth",
```
Comment (I-am-Erk) [inline data/json/items/armor/undergarment.json]: loincloths likewise aren't really undergarments.

File: data/json/items/armor/undergarment.json [hunk 6]
```diff
@@ -0,0 +1,743 @@
...
+    "symbol": "[",
+    "looks_like": "briefs",
+    "color": "brown",
+    "covers": [ "LEGS" ],
+    "coverage": 5,
+    "material_thickness": 1,
+    "flags": [ "VARSIZE", "SKINTIGHT", "OVERSIZE" ]
+  },
+  {
+    "id": "loincloth_fur",
```
Comment (I-am-Erk) [inline data/json/items/armor/undergarment.json]: as above

File: data/json/items/armor/undergarment.json [hunk 7]
```diff
@@ -0,0 +1,743 @@
...
+    "looks_like": "loincloth_leather",
+    "color": "brown",
+    "covers": [ "LEGS" ],
+    "coverage": 5,
+    "warmth": 5,
+    "material_thickness": 1,
+    "flags": [ "VARSIZE", "SKINTIGHT", "OVERSIZE" ]
+  },
+  {
+    "id": "loincloth_leather",
```
Comment (I-am-Erk) [inline data/json/items/armor/undergarment.json]: as above

File: data/json/items/armor/undergarment.json [hunk 8]
```diff
@@ -0,0 +1,743 @@
...
+    "symbol": "[",
+    "looks_like": "loincloth",
+    "color": "brown",
+    "covers": [ "LEGS" ],
+    "coverage": 5,
+    "material_thickness": 2,
+    "flags": [ "VARSIZE", "SKINTIGHT", "OVERSIZE" ]
+  },
+  {
+    "id": "loincloth_wool",
```
Comment (I-am-Erk) [inline data/json/items/armor/undergarment.json]: as above

File: data/json/items/tool_armor.json [hunk 9]
```diff
@@ -1,4 +1,148 @@
...
+    "copy-from": "arm_splint",
+    "looks_like": "arm_splint",
+    "weight": "600 g",
+    "volume": "2 L",
+    "price": 25000,
+    "encumbrance": 140,
+    "flags": [ "OVERSIZE", "WATER_FRIENDLY", "OUTER", "SPLINT" ]
+  },
+  {
+    "id": "barrette",
```
Comment (I-am-Erk) [inline data/json/items/tool_armor.json]: this seems more like jewelry to me

**Comment by RarkGrames** [comment]
Yeah, well I wanted to dump every skintight clothes into it, but I don't know what is a good word for all the clothing articles that you wear under other stuff for comfort or warmth. It's not "underwear" for sure, but my vocabulary is not the greatest. You have a point on loincloth though.

**Comment by I-am-Erk** [comment]
I'd still put Barette under jewelry, because that's its main role. It's just coincidence that it also happens to have a tool function, it's not a tool that you wear really.

I would put the rest of those non-undergarments just into a trousers.json file, or whatever you've called it. I think that's where people would look for something like hot pants or leggings.

**Comment by RarkGrames** [comment]
Done.

**Pull request merged**